DigitalSensor.cpp

```diff
@@ -1,151 +1,36 @@
-
-    //Return a pointer to the private value
-    return sensorValue;
-};
-
-
-/*! 
- * @brief Get ID
- * Returns the ID of the sensor
- * 
- * @param None
- * @return sensorID (int) The ID of the sensor
- */
-int DigitalSensor::getId(){
-    return sensorID;
-};
-
-
-/*! 
- * @brief Get input pin
- * Returns the pin that the sensor is connected to
- * 
- * @param None
- * @return input pin (int) The pin of the sensor
- */
-int DigitalSensor::getPins(){
-    return inputPins[0];
-};
-
-
-/*! 
- * @brief Get wait time
- * Returns the time that the sensor waits between
- * reporting. In milliseconds
- * 
- * @param None
- * @return waitTime (int) The wait time of the sensor
- */
-int DigitalSensor::getWaitTime(){
-    return waitTime;
-};
-
-
-/*! 
- * @brief Get data length
- * Returns the length of the data array. For 
- * digital sensors this will always be 1
- * 
- * @param None
- * @return 1 (int) The length of the data
- */
-int DigitalSensor::getDataLength(){
-    return 1;
-};
-
-
-/*!
- * @brief Builds the data array for the CAN message.
- * 
- * @param torque (int) The torque value to be sent.
- * @param percent (int) The percent value to be sent.
- * 
- * @return (int*) The data array for the CAN message.
-*/
-int* DigitalSensor::buildData(int value) {
-    return new int[1]{value};
-};
-
-/*!
- * @brief Builds the error data array for the CAN message.
- * 
- * @param errorType (int) The type of error.
- * @param errorData (int*) The data associated with the error.
- * 
- * @return (int*) The error data array for the CAN message.
-*/
-int* DigitalSensor::buildError() {
-    return new int[1]{0};
+DigitalSensor::DigitalSensor(const uint32_t id, const bool criticality, const uint8_t pin, const uint32_t readInterval)
+{
+    this->id = id;
+    this->criticality = criticality;
+    this->readInterval = readInterval;
+    this->pin = pin;
 }
 
-/*!
- * @brief Transform the sensor data into new scale.
- * @param data (int) The data to be rescaled.
- * 
- * Digital Sensors will not need to be rescaled
- * 
- * @return -1 because this should not be used
- */
-int DigitalSensor::rescale(int data) {
-    //Transform data
-    return -1;
-};
-
-bool DigitalSensor::readyToCheck() {
-    return (waitTime <= int(millis() - previousUpdateTime));
-}
+SensorData DigitalSensor::read()
+{
+    lastRead = millis();
+    SensorData sensorData = SensorData(id, 1);
 
+    const uint8_t buf[1] = { digitalRead(pin) };
+    sensorData.setMsg(buf, 1);
 
-bool DigitalSensor::getCritical() {
-    return critical;
+    return sensorData;
 }
 
-bool DigitalSensor::plugTest() {
-    // Change pinMode twice
-    pinMode(inputPins[0], INPUT_PULLDOWN);
-    if (digitalRead(inputPins[0]) == 0) {
-        pinMode(inputPins[0], INPUT_PULLUP);
-        if(digitalRead(inputPins[0] == 0)) {
-            pinMode(inputPins[0], INPUT_PULLDOWN);
-            return true;
-        }
-    }
-    pinMode(inputPins[0], INPUT_PULLDOWN);
-    return false;
-}
```

Comment:
Same concern with the plugTest in analog as well. DC & ECU want these to determine overall health of car on boot

Reply:
Although this function don't work but a placeholder for when the guy on the team this issue is assigned to puts in the fix